src/mod_mam_muc.erl

```diff
@@ -770,53 +769,14 @@ report_issue(Reason, Stacktrace, Issue, #jid{lserver = LServer, luser = LUser},
 %% ----------------------------------------------------------------------
 %% Dynamic params module
 
-%% compile_params_module([
-%%      {add_archived_element, boolean()}
-%%      ])
-compile_params_module(Params) ->
-    CodeStr = params_helper(Params),
-    {Mod, Code} = dynamic_compile:from_string(CodeStr),
-    code:load_binary(Mod, "mod_mam_muc_params.erl", Code).
-
-params_helper(Params) ->
-    %% Try is_complete_message opt for backwards compatibility
-    {IsArchivableModule, IsArchivableFunction} =
-        case proplists:get_value(is_archivable_message, Params) of
-            undefined ->
-                case proplists:get_value(is_complete_message, Params) of
-                    undefined -> {mod_mam_utils, is_archivable_message};
-                    OldStyleMod -> {OldStyleMod, is_complete_message}
-                end;
-
-            Mod -> {Mod, is_archivable_message}
-        end,
+param(Host, Opt, Default) ->
+    mod_mam_utils:param(?MODULE, Host, Opt, Default).
+
+is_archivable_message(Host, Dir, Packet) ->
+    mod_mam_utils:call_is_archivable_message(?MODULE, Host, Dir, Packet).
+
+add_archived_element(Host) ->
+    mod_mam_utils:add_archived_element(?MODULE, Host).
 
-    Format =
-        io_lib:format(
-          "-module(mod_mam_muc_params).~n"
```

Comment:
I like the param module, because I can get all MAM related settings without thinking for default value of each option.
Can we have a function, that returns all possible MAM parameters with current values?

Something like

```erlang
mod_mam:get_parameters(Host) ->
[{add_archived_element, false}, {default_result_limit, 50}, ....]
```

Same for mod_mam_muc.

Reply:
> I like the param module, because I can get all MAM related settings without thinking for default value of each option.

True, that's something to improve.

> Can we have a function, that returns all possible MAM parameters with current values?

What would be the point of such function?

Reply:
I'm hesitant to implement such functionality for MAM modules. If we want more explicit support for defaults, we should add it to gen_mod.

Reply:
Obviously sans defaults, the runtime config should just be fetched with `gen_mod:get_module_opts`